### ISSUE TITLE
The loss is Nan

### ISSUE THREAD

**Issue by abc8350712** [shekkizh/FCN.tensorflow#17]
I train my dataset which has only two class,so I set the NUM_CLASSES as 2,and the loss turned out to be Nan.I change the NUM _CLASSED to 3 or 151 without changing my dataset,and it can work.
I'm very confused with this,please help me.

I have tried to decrease the lr to le-7 and le-8,but it didn't work.

**Comment by xtudbxk** [comment]
It's funny. And I also train the code on my dataset. When the NUM_OF_CLASSES is less than 60, the loss is Nan. And when I change the NUM_OF_CLASSES to 61, it works. 
After some experiments, I found some results:
First, the direct reason is because the values of conv1_1 is Nan when the NUM_OF_CLASSES is less than 60. But the input image is not Nan. Second, what really matters is the NUM_OF_CLASSES in the third transpose convolution. That's saying it's no matter how you change the NUM_OF_CLASSES in the first transpose convolution.
But unfortunatly, I didn't find the reson and don't know how to fix it.
Do you fix it ?
Thanks,

**Comment by Exuro889** [comment]
Make sure your class labels are 0 and 1. I'm guessing the reason 151 works is because your label values fit within that range. The model afaik is trying to allocate a position in a one hot vector for each pixel, so if your label is outside the range of that vector then it will go nan.

**Comment by rayhou0710** [comment]
Agree with @Exuro889 . Just double check you label.

**Comment by abc8350712** [comment]
@xtudbxk  I have found the problem.I checked my dataset and surprisingly  the max value of my label is 2.No wonder that 2 classes didn't work.I strongly suggest you checking your label again to make sure the num of labels is 60.
Meanwhile,thanks you @rayhou0710 @Exuro889

**Comment by xtudbxk** [comment]
@abc8350712 @Exuro889 
Thanks for your help, I just get the loss which is not nan through your ways.

**Comment by shekkizh** [comment]
Sorry for the late response. The above comment by @Exuro889  does solve the problem.
Just fyi in the documentation for [tf.nn.sparse_softmax_cross_entropy_with_logits](https://www.tensorflow.org/api_docs/python/tf/nn/sparse_softmax_cross_entropy_with_logits) you will notice the labels are to be in range [0, num_classes). This is the same conclusion made in the above comments. This is just how the function is implemented.

**Comment by SoulGuedria** [comment]
Hi,
I'm trying to use my own dataset. I only have 2 classes and all my annotation images are grayscale. I changed NUM_OF_CLASSES to 2 but i still have the same problem.
I can't find how to check my labels. Is there any annotation file ? What changes have to be made in it?

Thanks a lot :)

**Comment by SoulGuedria** [comment]
@shekkizh @abc8350712 @Exuro889 
How can i label the classes in my own dataset please ? Is there any annotation file ? Thanks a lot :)
